helix-core/src/syntax.rs

```diff
@@ -308,8 +308,13 @@ impl Loader {
             .and_then(|n| n.to_str())
             .and_then(|file_name| self.language_config_ids_by_file_type.get(file_name))
             .or_else(|| {
-                path.extension()
-                    .and_then(|extension| extension.to_str())
+                // we can't use the `extension` methods here because it ignore
+                // empty file name with only an extension like `.bashrc`.
+                path.file_name()
+                    .map(|file_name| file_name.to_str())
+                    .flatten()
+                    .map(|file_name| file_name.rsplit('.').next())
+                    .flatten()
```

Comment:
IIRC currently `.gitconfig` already works since that is used for tree-sitter-git-config.

Reply:
Huum from what I see, no, it doesn’t work with git-config on master currently 🤔

Reply:
I see highlights working on master for `.gitmodules`/`.gitconfig` (i.e. `cargo run .gitmodules`). That's not working for you?

Reply:
Do you need to update your submodules?

Reply:
I’m not really good with submodules, is there a way to update my submodule more than that?
https://asciinema.org/a/tmOpnwQAFWhcWkJD3dFQJoIed

Reply:
Ah ok, I think if you `cargo clean -p helix-syntax` after the submodule update and then `cargo run ~/.gitconfig`, that might do it. You can also add a `--force` on the end of that submodules command iirc, and that forces them to be checked out again at whatever revision git thinks they should be.

btw the animal emojis on your shell are awesome! :smile:

Reply:
Yeah I don't think this needs to be changed, `file_name` will return `.bashrc` so specifying `".bashrc"` in TOML should be enough.